### ISSUE TITLE
spec.driver.env/envFrom is not working even when webhook is enabled

### ISSUE THREAD

**Issue by shinen** [kubeflow/spark-operator#1108]
I deployed a spark operator with manifest/spark-operator-with-webhook.yaml, with -enable-webhook=true.
However using env or envFrom doesn't inject the environment variables to driver and executor.

What's the problem?

Following is how I define env in my spark application yml. I have tested deploying each of them separately. 

Only envVars(which will be deprecated) works.
>
    driver:
      env:   
        - name: ENV_TWO
          value: hello
        - name: ENV_TWO
             valueFrom:
               secretKeyRef:
                 name: secretenv
                 key: TEST
      envFrom:
        - secretRef:
            name: secretenv
      envVars:
        TEST_ENVVARS: test

**Comment by sakshi-bansal** [comment]
env is not working for me as well. The spark application config shows it: driver:
env:
- name: ENV1
value: VAL1

But the env variables are not created inside the pods.

**Comment by shinen** [comment]
Anyone can help?

**Comment by nooshin-mirzadeh** [comment]
I faced the same problem, any solution?

**Comment by kz33** [comment]
hi,all 
I have not encountered this kind of problem, but I am happy to help you troubleshoot what happened。

create secret env `kubectl create -f spark-secret-env.yaml`,the content is as follows:
```
apiVersion: v1
kind: Secret
metadata:
  name: spark-secret-env
  namespace: szww
type: Opaque
data:
  password: cGFzc3dvcmQK
  username: YWRtaW4=
```


Now，i create sparkapplication with the following yaml
```
apiVersion: "sparkoperator.k8s.io/v1beta2"
kind: SparkApplication
metadata:
  name: spark-pi-test-env
  namespace: szww
spec:
  type: Scala
  mode: cluster
  image: "gcr.io/spark-operator/spark:v3.0.0"
  imagePullPolicy: Always
  mainClass: org.apache.spark.examples.SparkPi
  mainApplicationFile: "local:///opt/spark/examples/jars/spark-examples_2.12-3.0.0.jar"
  sparkVersion: "3.0.0"
  arguments:
    - "10000"
  restartPolicy:
    type: Never
  volumes:
    - name: "test-volume"
      hostPath:
        path: "/tmp"
        type: Directory
  driver:
    cores: 1
    coreLimit: "1200m"
    memory: "512m"
    env:
    - name: "ENV1"
      value: "VAL1"
    - name: "USER"
      valueFrom:
        secretKeyRef:
          name: spark-secret-env
          key: username
    envFrom:
    - secretRef:
        name: spark-secret-env
    labels:
      version: 3.0.0
    serviceAccount: spark
    volumeMounts:
      - name: "test-volume"
        mountPath: "/tmp"
  executor:
    cores: 1
    instances: 1
    memory: "512m"
    env:
    - name: "ENV1"
      value: "VAL1"
    - name: "USER"
      valueFrom:
        secretKeyRef:
          name: spark-secret-env
          key: username
    envFrom:
      - secretRef:
          name: spark-secret-env
    labels:
      version: 3.0.0
    volumeMounts:
      - name: "test-volume"
        mountPath: "/tmp"

```

Then I run the following command to enter the container, in order to see if the environment variables were injected.
```
# kubectl -n szww exec -it spark-pi-test-env-driver bash
```

I entered the container and was able to get the env
```
185@spark-pi-test-env-driver:~/work-dir$ echo $ENV1
VAL1
185@spark-pi-test-env-driver:~/work-dir$ echo $USER
admin
185@spark-pi-test-env-driver:~/work-dir$ echo $username
admin
185@spark-pi-test-env-driver:~/work-dir$ echo $password
password
```

**Comment by kz33** [comment]
Hi, All
I think I found the answer, you can try to use this option (`enable-webhook=true`)when running operator

**Comment by vvavepacket** [comment]
@kz33 Tried your approach and still enviroment variables are not making their way into their containers.

@nooshin-mirzadeh @shinen  @sakshi-bansal 

Is this still a valid issue with you?

**Comment by vvavepacket** [comment]
This is now working.. Just make sure you update your latest helm chart.

**Comment by mtaron** [comment]
Hm, I am still hitting this on the latest helm chart.

**Comment by 5RK7N** [comment]
I am having the same problem with chart version 1.0.7, running on EKS 1.18. As workaround I set it directly to `Sparkconfig`
```
spec:
  sparkConf:
    "spark.kubernetes.driverEnv.[EnvironmentVariableName]": "value"
```

**Comment by chaudhryfaisal** [comment]
**env** , and **envFrom** not working with chart 1.1.7 and operator v1beta2-1.2.2-3.0.0

installed 
```bash
helm upgrade --install --version 1.0.7 sparkoperator spark-operator/spark-operator \
		--namespace spark-operator --set sparkJobNamespace=spark-apps,webhook.enable=true \
		--set operatorImageName=spark-operator/spark-operator,operatorVersion=v1beta2-1.2.2-3.0.0 \
		--set imagePullPolicy=IfNotPresent
```

**Comment by debu99** [comment]
eks 1.18 also not working

**Comment by swill** [comment]
I spent many hours trying to troubleshoot this issue today.  I posted my findings here: https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/1229#issuecomment-827896078

TLDR; While I could not get the `env` or `envFrom` methods to work, I was able to get unblocked (for now) using `envSecretKeyRefs`.
